x-pack/plugins/integration_assistant/public/components/create_integration/create_integration_assistant/steps/data_stream_step/sample_logs_input.tsx

```diff
@@ -177,7 +196,25 @@ export const SampleLogsInput = React.memo<SampleLogsInputProps>(({ integrationSe
           samplesFormat,
         });
       };
-      setIsParsing(true);
+
+      reader.onerror = function () {
+        const message = reader.error?.message;
+        if (message) {
+          setSampleFileError(i18n.LOGS_SAMPLE_ERROR.CAN_NOT_READ_WITH_REASON(message));
+        } else {
+          setSampleFileError(i18n.LOGS_SAMPLE_ERROR.CAN_NOT_READ);
+        }
+      };
+
+      reader.onabort = function () {
+        const message = reader.error?.message;
+        if (message) {
+          setSampleFileError(i18n.LOGS_SAMPLE_ERROR.CAN_NOT_READ_WITH_REASON(message));
+        } else {
+          setSampleFileError(i18n.LOGS_SAMPLE_ERROR.CAN_NOT_READ);
+        }
+      };
```

Comment:
Extract this function out and call it in both `onerror` and `onabort`

Reply:
I've tried a direct assignment. Looks good?

Reply:
I understood this. But what is the benefit of this approach over the one I adopted?

Reply:
Yeah.. I meant to reuse the function.. You might have done this in a similar way in the latest [commit](https://github.com/elastic/kibana/pull/191310/commits/eb0d7effbdf6459b564437559889aa75894d80f3).

Reply:
I was more talking about this for better readability

```ts
// Define a common error handler function
const handleReaderError = () => {
  const errorMessage = reader.error?.message;
  const errorKey = errorMessage 
    ? i18n.LOGS_SAMPLE_ERROR.CAN_NOT_READ_WITH_REASON(errorMessage)
    : i18n.LOGS_SAMPLE_ERROR.CAN_NOT_READ;

  setSampleFileError(errorKey);
};

// Assign the same error handler to both onerror and onabort
reader.onerror = handleReaderError;
reader.onabort = handleReaderError;

```

Reply:
New function looks to me to be a very marginal improvement, but sure.
I don't think ternary operator here helps readability.

Reply:
Sure.. This is just a suggestion. Feel free to choose how you want the function to be.. And yeah, this is a nit

Reply:
I've implemented the function though not the ternary operator. Do you think it looks better with the ternary operator?

Reply:
I am OK either way. I leave this decision to you. 

I would use ternary operator if it is always `a or b` and an if condition if `ever there could be a new condition c`

Reply:
Awesome. If this works for you, please resolve the comment.
I always leave the comment resolution to the person who created it to avoid misunderstandings.